tests/test_add_model.py

```diff
@@ -0,0 +1,423 @@
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+from flask_restful import fields
+
+from flask_restful_swagger import swagger
+
+###############################################################################
+# Copy setup objects from examples/basic.py
+###############################################################################
+
+
+class MockBasicObject:
+    pass
+
+
+class MockBasicWithSwaggerMetadata1:
+    swagger_metadata = {"an_enum": {"enum": ["one", "two", "three"]}}
+
+class MockBasicWithSwaggerMetadata2:
+    def __init__(self, arg1, an_enum):
+        pass
+
+    swagger_metadata = {"an_enum": {"enum": ["one", "two", "three"]}}
+
+
+class TodoItem:
+    """This is an example of a model class that has parameters in its constructor
+    and the fields in the swagger spec are derived from the parameters
+    to __init__.
+    In this case we would have args, arg2 as required parameters and arg3 as
+    optional parameter.
+    """
+
+    def __init__(self, arg1, arg2, arg3="123"):
+        pass
+
+
+class ModelWithResourceFieldsNoRequired:
```

Comment:
Same thing, if we could prefix all this objects with something that identifies their purpose that might be helpful for folks reading the test.